gadgets/python/PythonGadget.cpp

```diff
@@ -1,6 +1,65 @@
 #include "PythonGadget.h"
 
 namespace Gadgetron{
-  GADGET_FACTORY_DECLARE(AcquisitionPythonGadget)
-  GADGET_FACTORY_DECLARE(ImagePythonGadget)
+  int PythonGadget::process(ACE_Message_Block* mb)
+  {
+    GadgetContainerMessage<ISMRMRD::AcquisitionHeader>* hma = AsContainerMessage<ISMRMRD::AcquisitionHeader>(mb);
+    if (hma) {
+      GadgetContainerMessage< hoNDArray< std::complex<float> > >* dmb = AsContainerMessage< hoNDArray< std::complex<float> > >(hma->cont());
+      if (!dmb) {
+	GERROR("Error converting data array from message block for Acquisition\n");
+	hma->release();
+	return GADGET_FAIL;;
+      }
+      GadgetContainerMessage< ISMRMRD::MetaContainer>* mmb = AsContainerMessage< ISMRMRD::MetaContainer >(dmb->cont());
+      this->process(hma,dmb,mmb);
+    } else {
+      GadgetContainerMessage<ISMRMRD::ImageHeader>* hmi = AsContainerMessage<ISMRMRD::ImageHeader>(mb);
+      if (!hmi) {
+	if (pass_on_undesired_data.value()) {
+	  return this->next()->putq(mb);
+	} else {
+	  GERROR("This is neither an acquisition or an image. Something is wrong here");
+	  mb->release();
+	  return GADGET_FAIL;
+	}
+      }
```

Comment:
@inati, it is possible in theory, but implementation for other data types is not supported. If it is not an acquisition header, we check if it is an image header, if it is not, it is passed on to the next gadget (if pass_on_undesired_data is set) or it is an error. 

If we want to add support for more data types here, some restructuring might be needed.